### ISSUE TITLE
Fails to build with "narrowing" warning

### ISSUE THREAD

**Issue by nyov** [icculus/smpeg#1]
Current master fails to build with "narrowing conversion" warning (on GCC 8 here).

```make
audio/hufftable.cpp:587:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘unsigned int’ inside { } [-Wnarrowing]
 };
 ^
audio/hufftable.cpp:587:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘unsigned int’ inside { } [-Wnarrowing]
audio/hufftable.cpp:587:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘unsigned int’ inside { } [-Wnarrowing]
audio/hufftable.cpp:587:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘unsigned int’ inside { } [-Wnarrowing]
audio/hufftable.cpp:587:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘unsigned int’ inside { } [-Wnarrowing]
audio/hufftable.cpp:587:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘unsigned int’ inside { } [-Wnarrowing]
make: *** [Makefile:843: hufftable.lo] Error 1
```

Workaround is to set the `-Wno-error=narrowing` flag with gcc, but it's probably not correct as a solution.

Thanks for having this on Github.

**Comment by ebachard** [comment]
Hello,

FYI, I proposed a PR with a working fix. See : https://github.com/icculus/smpeg/pull/2

Thanks in advance to confirm it works (or not) for you.

**Comment by icculus** [comment]
I merged #2, should resolve this (thanks!).

Let me know if other parts are still broken!
